### PR TITLE
Remove the extra codemirror plugins

### PR DESCRIPTION
**Pull request by blink1073**
cf https://github.com/jupyterlab/jupyterlab/pull/1140#issuecomment-264646011

- [x] Removes `src/console/codemirror` by intercepting 'enter' key using `onKeyDown`
- [x] Removes `src/notebook/codemirror` by letting the codemirror implementation set the default `matchBrackets` and `autoCloseBrackets` based on mimetype, and moving the theme to `src/codemirror`.
- [x] Update examples and tests

The only other places outside of `src/codemirror` where we use `Codemirror` are to get the default language for a document model in `src/docregistry/default.ts` and to highlight markdown cells in `src/renderers/widget.ts`.

### CODE REVIEW

File: src/notebook/plugin.ts [hunk 1]
```diff
@@ -142,7 +146,11 @@ const plugin: JupyterLabPlugin<INotebookTracker> = {
 /**
  * Activate the notebook handler extension.
  */
-function activateNotebookHandler(app: JupyterLab, registry: IDocumentRegistry, services: IServiceManager, rendermime: IRenderMime, clipboard: IClipboard, mainMenu: IMainMenu, palette: ICommandPalette, inspector: IInspector, renderer: NotebookPanel.IRenderer, state: IStateDB, layout: ILayoutRestorer): INotebookTracker {
+function activateNotebookHandler(app: JupyterLab, registry: IDocumentRegistry, services: IServiceManager, rendermime: IRenderMime, clipboard: IClipboard, mainMenu: IMainMenu, palette: ICommandPalette, inspector: IInspector, editorServices: IEditorServices, state: IStateDB, layout: ILayoutRestorer): INotebookTracker {
+
+  const notebookRenderer = new Notebook.Renderer({ editorServices });
```
Comment (blink1073) [inline src/notebook/plugin.ts]: Okay, then I will make the default `IRenderer` plugins use `IEditorServices` only, but allow Monaco to provide the more advanced `IRenderer`.

File: src/console/content.ts [hunk 2]
```diff
@@ -837,4 +824,23 @@ namespace Private {
...
+    /**
+     * Create a new cell editor for the widget.
+     */
+    createCellEditor(): ICellEditorWidget {
+      let widget = super.createCellEditor();
+      // Suppress the default "Enter" key handling.
+      let cb = (editor: CodeEditor.IEditor, event: KeyboardEvent) => {
+        return event.keyCode === 13;  // Enter;
+      };
+      widget.editor.onKeyDown = cb;
```
Comment (akosyakov) [inline src/console/content.ts]: Won't it be removed by the cell editor widget?

Comment (blink1073) [inline src/console/content.ts]: Good point, what we really need is to allow for multiple keyboard event filters to be registered on the editor.

**Comment by blink1073** [comment]
Updated, we now have `addKeydownHandler(handler: CodeEditor.KeydownHandler): IDisposable`.

**Pull request merged**